src/main/java/org/commcare/cases/instance/IndexedFixtureChildElement.java

```diff
@@ -0,0 +1,74 @@
+package org.commcare.cases.instance;
+
+import org.commcare.cases.model.StorageIndexedTreeElementModel;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+
+/**
+ * Child TreeElement of an indexed fixture whose data is loaded from a DB.
+ *
+ * i.e. 'product' of "instance('product-list')/products/product"
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class IndexedFixtureChildElement extends StorageBackedChildElement<StorageIndexedTreeElementModel> {
+    private TreeElement empty;
+
+    protected IndexedFixtureChildElement(StorageInstanceTreeElement<StorageIndexedTreeElementModel, ?> parent,
+                                         int mult, int recordId) {
+        super(parent, mult, recordId, parent.getName(), parent.getChildHintName());
+    }
+
+    /**
+     * Template constructor (For elements that need to create reference nodesets but never look up values)
+     */
+    private IndexedFixtureChildElement(StorageInstanceTreeElement<StorageIndexedTreeElementModel, ?> parent) {
+        super(parent, TreeReference.INDEX_TEMPLATE,
+                TreeReference.INDEX_TEMPLATE, parent.getName(),
+                parent.getChildHintName());
+
+        StorageIndexedTreeElementModel modelTemplate = parent.getModelTemplate();
+        empty = modelTemplate.getRoot();
+        empty.setMult(TreeReference.INDEX_TEMPLATE);
+        // NOTE PLM: do we need to do more to convert a regular TreeElement into a template?
+    }
+
+    @Override
+    protected TreeElement cache() {
```

Comment:
I think it caches the result of deriving the underlying TreeElement from the current object